external/storm-kafka-client/src/test/java/org/apache/storm/kafka/KafkaUnit.java

```diff
@@ -0,0 +1,100 @@
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package org.apache.storm.kafka;
+
+import kafka.admin.AdminUtils;
+import kafka.admin.RackAwareMode;
+import kafka.server.KafkaConfig;
+import kafka.server.KafkaServer;
+import kafka.utils.*;
+import kafka.zk.EmbeddedZookeeper;
+import org.I0Itec.zkclient.ZkClient;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Properties;
+
+public class KafkaUnit {
+    private KafkaServer kafkaServer;
+    private EmbeddedZookeeper zkServer;
+    private ZkUtils zkUtils;
+    private KafkaProducer<String, String> producer;
+    private static final String ZK_HOST = "127.0.0.1";
+    private static final String KAFKA_HOST = "127.0.0.1";
+    private static final int KAFKA_PORT = 9092;
+
+    public KafkaUnit() {}
+
+    public void setUp() throws IOException {
+        // setup ZK
+        zkServer = new EmbeddedZookeeper();
+        String zkConnect = ZK_HOST + ":" + zkServer.port();
+        ZkClient zkClient = new ZkClient(zkConnect, 30000, 30000, ZKStringSerializer$.MODULE$);
+        zkUtils = ZkUtils.apply(zkClient, false);
+
+        // setup Broker
+        Properties brokerProps = new Properties();
+        brokerProps.setProperty("zookeeper.connect", zkConnect);
+        brokerProps.setProperty("broker.id", "0");
+        brokerProps.setProperty("log.dirs", Files.createTempDirectory("kafka-").toAbsolutePath().toString());
+        brokerProps.setProperty("listeners", String.format("PLAINTEXT://%s:%d", KAFKA_HOST, KAFKA_PORT));
+        KafkaConfig config = new KafkaConfig(brokerProps);
+        Time mock = new MockTime();
+        kafkaServer = TestUtils.createServer(config, mock);
+    }
+
+    public void tearDown() {
+        kafkaServer.shutdown();
+        zkUtils.close();
+        zkServer.shutdown();
+    }
+
+    public void createTopic(String topicName) {
+        AdminUtils.createTopic(zkUtils, topicName, 1, 1, new Properties(), RackAwareMode.Disabled$.MODULE$);
+    }
+
+    public int getKafkaPort() {
+        return KAFKA_PORT;
+    }
+
+    public void createProducer() {
+        Properties producerProps = new Properties();
+        producerProps.setProperty("bootstrap.servers", KAFKA_HOST + ":" + KAFKA_PORT);
+        producerProps.setProperty("key.serializer", "org.apache.kafka.common.serialization.StringSerializer");
+        producerProps.setProperty("value.serializer", "org.apache.kafka.common.serialization.StringSerializer");
+        producer = new KafkaProducer<>(producerProps);
+    }
+
+    public void createProducer(Class keySerializer, Class valueSerializer) {
```

Comment:
Sorry, I misunderstood the code. You can put it back if you want. You might want to change the signature to `createProducer(Serializer, Serializer)` and use this constructor though https://kafka.apache.org/090/javadoc/org/apache/kafka/clients/producer/KafkaProducer.html#KafkaProducer(java.util.Properties,%20org.apache.kafka.common.serialization.Serializer,%20org.apache.kafka.common.serialization.Serializer)